core-services/src/main/java/org/zalando/nakadi/service/publishing/NakadiKpiPublisher.java

```diff
@@ -103,23 +116,30 @@ public void publish(final Supplier<KPIEvent> kpiEventSupplier) {
     }
 
     private GenericRecord buildMetaDataGenericRecord(
-            final String eventType, final Schema schema, final String version) {
-        return buildMetaDataGenericRecord(eventType, schema, version, "unknown");
+            final String eventType, final Schema metadataSchema,
+            final byte metadataSchemaVersion, final String etSchemaVersion) {
+        return buildMetaDataGenericRecord(eventType, metadataSchema, metadataSchemaVersion, etSchemaVersion, "unknown");
     }
 
     private GenericRecord buildMetaDataGenericRecord(
-            final String eventType, final Schema schema, final String version, final String user) {
+            final String eventTypeName, final Schema metadataSchema, final byte metadataSchemaVersion,
+            final String etSchemaVersion, final String user) {
         final long now = System.currentTimeMillis();
-        return new GenericRecordBuilder(schema)
+        final var eventType = this.eventTypeCache.getEventType(eventTypeName);
+        final var partitions = timelineService.getTopicRepository(eventType)
+                .listPartitionNames(timelineService.getActiveTimeline(eventType).getTopic());
+        final var metadata = new GenericRecordBuilder(metadataSchema)
                 .set("occurred_at", now)
                 .set("eid", uuidGenerator.randomUUID().toString())
                 .set("flow_id", FlowIdUtils.peek())
-                .set("event_type", eventType)
-                .set("partition", "0") // fixme avro
+                .set("event_type", eventTypeName)
                 .set("received_at", now)
-                .set("schema_version", version)
+                .set("schema_version", etSchemaVersion)
                 .set("published_by", user)
+                .set("partition", randomPartitionStrategy.calculatePartition(partitions))
```

Comment:
So we actually bypass the Avro question in the end?

Reply:
We resolved this conversation offline